src/commands/cli/validateSwaCliInstalled.ts

```diff
@@ -0,0 +1,59 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { MessageItem } from 'vscode';
+import { callWithTelemetryAndErrorHandling, DialogResponses, IActionContext } from 'vscode-azureextensionui';
+import { installSwaCliUrl } from '../../constants';
+import { localize } from '../../utils/localize';
+import { openUrl } from '../../utils/openUrl';
+import { getInstalledSwaCliVersion } from './getInstalledSwaCliVersion';
+import { installOrUpdateSwaCli } from './installOrUpdateSwaCli';
+import { hasNpm } from './validateSwaCliIsLatest';
+
+export async function validateSwaCliInstalled(context: IActionContext, message: string): Promise<boolean> {
+    let input: MessageItem | undefined;
+    let installed: boolean = false;
+    const install: MessageItem = { title: localize('install', 'Install') };
+
+    await callWithTelemetryAndErrorHandling('staticWebApps.validateSwaCliInstalled', async (innerContext: IActionContext) => {
+        innerContext.errorHandling.suppressDisplay = true;
+
+        const installedVersion: string | null = await getInstalledSwaCliVersion();
+
+        // No minimum version for now.
+        if (installedVersion) {
+            installed = true;
+        } else {
+            const items: MessageItem[] = [];
+            const isNpmInstalled = await hasNpm();
+            if (isNpmInstalled) {
+                items.push(install);
+            } else {
+                items.push(DialogResponses.learnMore);
+            }
+
+            // See issue: https://github.com/Microsoft/vscode-azurefunctions/issues/535
+            input = await innerContext.ui.showWarningMessage(message, { modal: true }, ...items);
+
+            innerContext.telemetry.properties.dialogResult = input.title;
+
+            if (input === install) {
+                await installOrUpdateSwaCli(context);
+                installed = true;
+            } else if (input === DialogResponses.learnMore) {
+                await openUrl(installSwaCliUrl);
+            }
+        }
+    });
+
+    // validate that SWA CLI was installed only if user confirmed
+    if (input === install && !installed) {
+        if (await context.ui.showWarningMessage(localize('failedInstallSwaCli', 'The Azure Static Web Apps CLI installation has failed and will have to be installed manually.'), DialogResponses.learnMore) === DialogResponses.learnMore) {
```

Comment:
`IAzureUserInput.showWarningMessage` takes a `learnMoreLink` parameter in the options, so you can just pass that `installSwaCliUrl` and don't need to check for the `DialogResponse` or open the URL yourself